packages/components/pie-thumbnail/src/index.ts

```diff
@@ -27,21 +28,48 @@ export class PieThumbnail extends LitElement implements ThumbnailProps {
     @property({ type: String })
     public alt = defaultProps.alt;
 
+    @property({ type: Boolean })
+    public disabled = defaultProps.disabled;
+
+    @property({ type: Boolean })
+    public hasPadding = defaultProps.hasPadding;
+
+    @property({ type: String })
+    public backgroundColor = defaultProps.backgroundColor;
+
+    @property({ type: Object })
+    public placeholder: ThumbnailProps['placeholder'];
+
+    private handleImageError = () => {
+        if (this.placeholder?.src) this.src = this.placeholder.src as string;
+        if (this.placeholder?.alt) this.alt = this.placeholder.alt as string;
+    };
+
     render () {
         const {
             variant,
             src,
             alt,
+            disabled,
+            hasPadding,
+            backgroundColor,
+            handleImageError,
         } = this;
 
         const wrapperClasses = {
             'c-thumbnail': true,
             [`c-thumbnail--${variant}`]: true,
+            'c-thumbnail--disabled': disabled,
+            'c-thumbnail--padding': hasPadding,
+        };
+
+        const wrapperStyles = {
+            backgroundColor,
         };
 
         return html`
-            <div data-test-id="pie-thumbnail" class="${classMap(wrapperClasses)}">
-                <img data-test-id="pie-thumbnail-img" src="${src}" class="c-thumbnail-img" alt="${alt}">
+            <div data-test-id="pie-thumbnail" class="${classMap(wrapperClasses)}" style="${styleMap(wrapperStyles)}">
```

Comment:
Sorry but I don't think we can allow this. It looks like a direct injection of raw CSS into the style tag. I think a user could put anything in here and that poses a security risk.

I think we should only let them select from a predefined list of options. Or something else more restricted.

Reply:
we updated to follow the variants approach as discussed offline.